src/v/datalake/serde_parquet_writer.h

```diff
@@ -7,21 +7,34 @@
 namespace datalake {
 class serde_parquet_writer : public parquet_ostream {
 public:
-    explicit serde_parquet_writer(serde::parquet::writer writer)
-      : _writer(std::move(writer)) {}
+    explicit serde_parquet_writer(
+      serde::parquet::writer writer, writer_mem_tracker& mem_tracker)
+      : _writer(std::move(writer))
+      , _mem_tracker(mem_tracker) {}
     ss::future<writer_error>
-      add_data_struct(iceberg::struct_value, size_t) final;
+    add_data_struct(iceberg::struct_value, size_t, ss::abort_source&) final;
+
+    size_t buffered_bytes() const final;
+
+    size_t flushed_bytes() const final;
```

Comment:
Should we just amend the interface of the parquet writer to be the same? So we don't have to duplicate the state in both layers?

Reply:
yeah makes sense to me. will do in a follow up so I can boy scout it a little bit.